### PR TITLE
Adding module website_twitter_js_self_hosted

### PR DESCRIPTION
**Pull request by gfcapalbo**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Maybe this one can fit better in OCA/social? Why the no_ext_links part?

**Comment by hbrunn** [comment]
check runbot and travis

File: website_twitter_no_ext_links/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Multiple Twitter Feeds and user stats with widget support"
+            "and no external links",
+    "version": "8.0.1.0.0",
+    "author": "Therp BV, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "category": "Website",
+    "summary": "",
```
Comment (hbrunn) [inline website_twitter_no_ext_links/__openerp__.py]: remove empty key or write a summary

File: website_twitter_no_ext_links/data/themes.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<openerp>
+	<data noupdate="1">
```
Comment (hbrunn) [inline website_twitter_no_ext_links/data/themes.xml]: fix indentation

File: website_twitter_no_ext_links/models/twitter_feed.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            "Content-Type",
+            "application/x-www-form-urlencoded;charset=UTF-8")
+        response = urllib2.urlopen(request)
+        response_str = response.read()
+        response_dict = json.loads(response_str)
+        return str(response_dict['access_token'])
+
+    @api.model
+    def get_userposts(self, feed):
+        self.ensure_one()
```
Comment (hbrunn) [inline website_twitter_no_ext_links/models/twitter_feed.py]: this makes no sense in combination with a `@api.model` function

File: website_twitter_no_ext_links/models/twitter_feed.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                            exists[0].write({'mentions': exists.mentions + 1})
+                        else:
+                            self.env['twitter.user'].create(entitydict)
+
+    def clear_tweets(self):
+        # Proposal maybe not delete but "archive"
+        old_tweets = self.env['twitter.tweet'].search(
+            [('feed_id', '=', self.id)]
+        )
+        old_tweets.sudo().unlink()
```
Comment (hbrunn) [inline website_twitter_no_ext_links/models/twitter_feed.py]: this allows every logged in user to delete all tweets. I don't think we want that

File: website_twitter_no_ext_links/README.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/website/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/website/issues/new?body=module:%20website_twitter_simple%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
+
```
Comment (hbrunn) [inline website_twitter_no_ext_links/README.rst]: find out if we're allowed to use the twitter logo in the first place, and if so, name your source here

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by hbrunn** [comment]
@pedrobaeza the reason we wrote this module is that we want our websites to be free of any js that pulls third party code from the net. This as well as privacy measure for the website users as as security measure in case one of those services ever is compromised. As such, it has less to do with social stuff but with the website

**Comment by pedrobaeza** [comment]
Thanks for the reasoning, Holger. Indeed it's a strong argument for the module. About the host in OCA/social, I still see no problem on moving to there, as the main topic is about social aspects. It doesn't mind that it happens on website or backend (mails and so on). See for example https://github.com/OCA/event/tree/8.0/website_event_contact, that is something for the website, but the transversal topic about event makes OCA/event a better hosting.

Anyways, we can hear another opinion on this if you want.

About the name, there are 2 objections:
- The plural on the name, which we at OCA avoids: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules
- It's not very self-descriptive. What about using as name `website_twitter_internal_js` or `website_twitter_js_self_hosted`?

**Comment by gfcapalbo** [comment]
@pedro, confirmed the absence of external js or other links. I do believe it belongs in website, afterall users look for snippets in /website mostly.

I am open to a name change, but  I think website_twitter_no_ext_links  could be ok (the plural!) if i add a summary/readme  description where i specify No_ext_links meaning better.

**Comment by hbrunn** [comment]
``` python

class Pedro(reviewbot):
    def review(self, mr):
        if any(mod.name.endswith('s') for mod in mr.modules):
            self.comment('No plurals!')
        # do review
```

;-)

**Comment by gfcapalbo** [comment]
no  plurals!
so can we agree to improve comprehension of this module by:

0) leaving in website (together with most a snippets)
1) renaming from website_twitter_no_ext_links   to website_twitter_no_ext_link  (no plural)
2) adding extra information in README explanation to describe what no_ext_link means.

?

**Comment by pedrobaeza** [comment]
@hbrunn :laughing: 

@gfcapalbo I won't block this anymore, so if you both decide that this is a good name and repository, go ahead.

**Comment by hbrunn** [comment]
I leave the decision to @gfcapalbo

**Comment by gfcapalbo** [comment]
As you both leave me the decision. I decide; I am thinking no_ext_links is confusing.
So i rename to website_twitter_js_self_hosted  (more clear) ,  add an extra line of explanation in README better and leave it in website repo.
i hope you like my decision, but you abdicated your vote.

**Comment by pedrobaeza** [comment]
:clap: :clap:

File: website_twitter_js_self_hosted/README.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===========================================================================
+Twitter Feeds and user stats with widget support and no external javascript
+===========================================================================
+
+A twitter extension that includes:
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:9]: s/twitter/Twitter

File: website_twitter_js_self_hosted/README.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===========================================================================
+Twitter Feeds and user stats with widget support and no external javascript
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:6]: s/javascript/JavaScript

File: website_twitter_js_self_hosted/README.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+===========================================================================
+
+A twitter extension that includes:
+
+    * No js that pulls third party code form the net.
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:11]: s/js/JS

Comment (pedrobaeza) [reply website_twitter_js_self_hosted/README.rst:11]: s/form/from

Comment (pedrobaeza) [reply website_twitter_js_self_hosted/README.rst:11]: Don't indent this, as the list looks weird in this case.

File: website_twitter_js_self_hosted/README.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    * Statistics on most mentioned users in every account.
+
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+
+# Go to Knowledge>Collaborative Content>Configure Twitter Account
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:25]: It should be `#.`

File: website_twitter_js_self_hosted/README.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+
+# Go to Knowledge>Collaborative Content>Configure Twitter Account 
+create a new Account (API Key and secret needed).
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:26]: add 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    the account needs to have:
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:27]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        1) a name
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:28]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+        2) a default theme
+        3) a user to fetch from
+
+# Go to Knowledge>Collaborative Content>Twitter Website Widget Themes
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:32]: `#.`

File: website_twitter_js_self_hosted/README.rst [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+# Go to Knowledge>Collaborative Content>Configure Twitter Account 
+create a new Account (API Key and secret needed).
+    the account needs to have:
+        1) a name
+        2) a default theme
+        3) a user to fetch from
+
+# Go to Knowledge>Collaborative Content>Twitter Website Widget Themes 
+and create a new theme, or just use one of the default ones.
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:33]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+   widget in the desired area.
+
+#. Choose in widget options twitter feed and theme.
+
+#. save
+
+Known issues / Roadmap
+======================
+
+TODO:
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:51]: Don't put this

File: website_twitter_js_self_hosted/README.rst [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    Create a widget
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:52]: Put bullets

File: website_twitter_js_self_hosted/README.rst [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/website/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/website/issues/new?body=module:%20website_twitter_simple%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:64]: Use new template for this.

File: website_twitter_js_self_hosted/__openerp__.py [hunk 18]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Multiple Twitter Feeds and user stats with widget support"
+            "and no external javascript",
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/__openerp__.py]: s/javascript/JavaScript

File: website_twitter_js_self_hosted/__openerp__.py [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+    "version": "8.0.1.0.0",
+    "author": "Therp BV, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "category": "Website",
+    "depends": [
+        'website',
+        'knowledge',
+        'website_parameterized_snippet'
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/__openerp__.py:14]: Put final comma

File: website_twitter_js_self_hosted/data/themes.xml [hunk 20]
```diff
@@ -0,0 +1,32 @@
+<openerp>
+	<data noupdate="1">
+
+		<record id="twitter_theme_dark" model="twitter.theme">
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/data/themes.xml:4]: Align correctly this

File: website_twitter_js_self_hosted/models/twitter_theme.py [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+class TwitterTheme(models.Model):
+    _name = 'twitter.theme'
+
+    name = fields.Char(
+        string='Twitter Theme Name',
+        required=True,
+    )
+
+    def _get_colors(self):
+        return [('#F0F8FF', 'AliceBlue'), ('#FAEBD7', 'AntiqueWhite'),
```
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/models/twitter_theme.py:16]: Isn't this list in any place that we need to recreate it?

**Pull request closed without merging**